### ISSUE TITLE
Marking issues in recorded macro

### ISSUE THREAD

**Issue by VladLazarovici** [notepad-plus-plus/notepad-plus-plus#1948]
**Have the following text:**
This is a text is want to keep
This is a text is want to remove
This is a text is want to keep
This is a text is want to keep
This is a text is want to remove
This is a text is want to keep

I.
Start recording a macro.
1. Go at the beginning of the text
1. Go to Find and Replace, Mark tab, check Bookmark line option and enter text _keep_ in Find what area
2. Press Mark all (leave window open) -> lines 1,3,4,6 are marked 
3. Go to Search - Bookmark - Remove unmarked lines -> lines 2 and 5 are removed
4. Press Clear all marks.  -> all marks are cleared
5. Close window and stop recording the macro.
Restore the original text and run the macro.

Expected: 
_This is a text is want to keep_ appears on 4 lines, all of which are **unmarked**,

Actual:
_This is a text is want to keep_ appears on 4 lines, all of which are **marked** 

II. 
**Have the same text.**
Start recording a macro.
1. Go at the beginning of the text
1. Go to Find and Replace, Mark tab, check Bookmark line option and enter text _keep_ in Find what area
2. Press Mark all (leave window open) -> lines 1,3,4,6 are marked 
3. Go to Search - Bookmark - Inverse bookmark -> lines 2 and 5 are marked, lines 1,3,4,6 are unmarked 
4. Go to Search - Bookmark - Remove bookmarked lines. -> Lines 2 and 5 are removed
5. Press Clear all marks.  -> all marks are cleared
6. Close window and stop recording the macro.
Restore the original text and run the macro.

Expected: 
**_This is a text is want to keep_ appears on 4 lines**, all of which are unmarked,

Actual:
**_This is a text is want to remove_ appears on 2 lines** 

System details: Intel i5 at 2.6 GHz, 4GB RAM, Windows 7 Ultimate 64 bit
Application version: Notepad++ v6.9.2

**Comment by sasumner** [comment]
Tested in version 7.5.8 on 26 September 2018.  Still exists.

@VladLazarovici 's case I problem seems to originate from the "Clear all marks" button press action not being recorded into the macro.  Indeed if you try to record ONLY that action in a macro, it results in Playback begin greyed out when you go to test it.

Case II adds the complication of inversing bookmarks.  That action also appears to not be recordable into a macro.  Again, if you try to record ONLY that action in a macro, it results in Playback begin greyed out when you go to test it.

**Comment by sasumner** [comment]
Followup on Case II:  Even though you can't record the inverse-bookmarks command (value is 43050) into a macro, the action will work if you hand-edit the macro in shortcuts.xml and insert the following into an existing macro where you want the inversing of bookmarks to occur:

<Action type="2" message="0" wParam="43050" lParam="0" sParam="" />

**Comment by MetaChuh** [comment]
please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html) and migrate or reinstall your plugins if you update from version 7.5.9 or below.
if your issue still persists, and it is still of relevance to you, please feel free to file a new issue as easy to understand and reproducible as possible.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)

**Comment by sasumner** [comment]
Reverified in Notepad++ 7.6.6.  STILL EXISTS, both case I and case II.

**Comment by sasumner** [comment]
I retested in Notepad++ 7.7 32-bits and recording a "Clear All Bookmarks" into a macro works successfully when playing back the macro.

Restesting "Inverse Bookmark" in 7.7 results in it still not being recordable (as described above). :(

**Comment by sasumner** [comment]
Inverse-bookmarking is not macro-recordable is because `IDM_SEARCH_INVERSEMARKS` is missing from the `if (_recordingMacro)` big switch statement in NppCommands.cpp.

**Comment by sasumner** [comment]
When I declared this, above:  `I retested` ... `and recording a "Clear All Bookmarks" into a macro works successfully when playing back the macro`, I introduced a replication error.

Notice I said `Clear All Bookmarks` whereas the OP's step was `Clear all marks`.  Clearing bookmarks (from the Search > Bookmarks menu) works for this application to remove bookmarks in a macro, whereas clearing all marks (pressing the button in the Mark window) does not.

Thus, as of this writing, current Notepad++ (7.7.1) still exhibits both the Case I and Case II bugs.  Deeply sorry for the confusion caused by this.
